cylc/flow/rundb.py

```diff
@@ -1085,3 +1085,21 @@ def select_jobs_for_datastore(
     def vacuum(self):
         """Vacuum to the database."""
         return self.connect().execute("VACUUM")
+
+    def message_in_db(self, submit, message, timestamp, itask):
+        """Has this message been logged in the DB (task_events table)?
+        """
+        task_name = itask.tokens['task']
+        cycle = itask.tokens['cycle']
+        stmt = r"""
+            SELECT EXISTS (
+                SELECT 1 FROM task_events WHERE
+                    name == ?
+                    AND cycle == ?
+                    AND message == ?
+                    AND submit_num == ?
+                    AND time == ?
+            )
+        """
+        stmt_args = [task_name, cycle, message, submit, timestamp]
+        return bool(self.connect().execute(stmt, stmt_args).fetchone())
```

Comment:
If using `SELECT EXISTS` it's returning either `(1,)` or `(0,)`
```suggestion
        return bool(self.connect().execute(stmt, stmt_args).fetchone()[0])
```